app/main.py

```diff
@@ -1,13 +1,47 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors":
+            [
+                {
+                    "line": err_single_file["line_number"],
+                    "column": err_single_file["column_number"],
+                    "message": err_single_file["text"],
+                    "name": err_single_file["code"],
+                    "source": "flake8"
+                } for err_single_file in errors
+
+            ],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [
+        {
+            "errors": [
+                {
+                    "line": single_error["line_number"],
+                    "column": single_error["column_number"],
+                    "message": single_error["text"],
+                    "name": single_error["code"],
+                    "source": "flake8"
+                } for single_error in file_errors
```

Comment:
The error formatting logic here is repeated from `format_single_linter_file`. It would be beneficial to extract this into a helper function to reduce code duplication.